packages/io-fims-common/src/domain/audit-event.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from "zod";
+
+import { userMetadataSchema } from "./user-metadata.js";
+
+export const rpParamsSchema = z.object({
+  client_id: z.string().min(1),
+  redirect_uri: z.string().min(1).optional(),
+  response_type: z.string().min(1).optional(),
+  scope: z.string().min(1).optional(),
+});
+
+export type RPParams = z.TypeOf<typeof rpParamsSchema>;
+
+export const auditEventSchema = z.object({
+  blobName: z.string().min(1),
+  idToken: z.string().min(1).optional(),
+  ipAddress: z.string().min(1).optional(),
+  rpParams: rpParamsSchema.optional(),
+  timestamp: z.number().optional(),
+  userData: userMetadataSchema.optional(),
```

Comment:
`idToken`, `ipAddress`, `rpParams`, `userData` ecc. are mandatory for the audit event, so they must not be `optional`

the rationale behind this choice is that you don't have all values at once when you write the blob the first time, right? for this use case you should use discriminated unions (aka sum types)

* https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html
* https://2ality.com/2020/06/type-guards-assertion-functions-typescript.html#checking-the-value-of-a-shared-property-(discriminated-unions)
* https://medium.com/axon-insights/mastering-sum-types-b588a3bc165b